File: 프로그래머스/lv0/120878. 유한소수 판별하기/유한소수 판별하기.js
function solution(a, b) {

  let getGCD = (num1, num2) => {
    let gcd = 1;
  
    for (let i = 2; i <= Math.min(num1, num2); i++) {
      if (num1 % i === 0 && num2 % i === 0) {
        gcd = i;
      }
    }
    return gcd;
  }

  // 최대 공약수 구하기
  const gcd = getGCD(a, b);

  let checkedValue = b / gcd;

  while (true) {
    if (checkedValue % 2 === 0) {
      checkedValue /= 2;
    } else if (checkedValue % 5 === 0) {
      checkedValue /= 5;
    } else {
      break;
    }
  }

  if (checkedValue === 1) {
    return 1;
  } else {
    return 2;
  }
}

File: 프로그래머스/lv0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    answer = 0
    # 기약분수 만들기
    for i in range(2, min(a, b) + 1):
        if a % i == 0 and b % i == 0:
            a /= i
            b /= i
    print(a, b)
    # 유한소수 판별
    while b % 2 == 0:
        b /= 2
    while b % 5 == 0:
        b /= 5
    if b == 1:
        return 1
    return 2
